### DISCUSSION TITLE
Programmer SW not responding

### DISCUSSION THREAD

**Discussion by Wispje** [danielnilsson9/bbs-fw#72] [Q&A] [answered]
Hi, I finished my BBS02 build and I'm trying to install the custom fw with the programming tool but I can't seem to make it work. The computer is recognizing the correct COM port but when hitting the 'Check MCU' button, nothing happens.

Editing the existing firmware with a config tool works just fine

Using the STC-ISP V6.88

**Comment by stts2** [comment]
Did you figure it out?

**Comment by Wispje** [comment]
Hi, no not yet. I don't know why the STC-ISP is not responding. Could it be that when installing I let it update to V6.90 but then cancelled it when I read that you should use V6.88. I don't know what it did during the installation but could this have currupted the V6.88 verison?

I'm also doing this from a macbook with bootcamp but I don't think this is an issue

**Comment by Wispje** [comment, marked as answer]
I also noticed the following steps in the manual:

_6. Power of your controller.
7. Click "Check MCU" and then power on your controller to verify._

But how do I power off and on the controller?

**Reply by stts2** [reply]
There you go. You did figure out how to do it. You just haven't done it yet. Yup, the motor responds to STC-ISP only on initial power up. Because after its powered up, its busy doing all the things in the firmware its already got. So you got the MAC and the modem cable and the new code on your screen ready to go. You can check the motor first by unplugging motor power then plugiing it back in. Then the motor will do the Check MCU command first thing before going off and running its code again. If all looks cool, then you can setup STC to burn in the new code. Then again, unplug motor power and plug back in. Prestoo Changeoo, your new code is being loaded in. Hopefully you compiled the code right and don't end up with a brick. :)

**Comment by Wispje** [comment]
I was off for some time but tried it again just now and it worked perfectly! Thanks for the help. I indeed did not know that you need to power off and on the whole system by connecting it to the battery. No idea why, but I thought the power of usb was what they meant in the guide. Thanks for the help @stts2
